backend/onedrive/onedrive.go

```diff
@@ -1503,7 +1503,75 @@ func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration,
 		fmt.Println(err)
 		return "", err
 	}
-	return result.Link.WebURL, nil
+
+	shareURL := result.Link.WebURL
+
+	// Convert share link to direct download link if target is not a folder
+	cnvFailMsg := "Don't know how to convert share link to direct link - returning the link as is"
+	if info.Folder != nil {
+		fs.Logf(f, cnvFailMsg)
+		return shareURL, nil
+	}
+
+	directURL := ""
+	segments := strings.Split(shareURL, "/")
+	switch f.driveType {
+	case driveTypePersonal:
+		// Method: https://stackoverflow.com/questions/37951114/direct-download-link-to-onedrive-file
+		if len(segments) != 5 {
+			fs.Logf(f, cnvFailMsg)
+			return shareURL, nil
+		}
+		enc := base64.StdEncoding.EncodeToString([]byte(shareURL))
+		enc = strings.ReplaceAll(enc, "/", "_")
+		enc = strings.ReplaceAll(enc, "+", "-")
+		enc = strings.ReplaceAll(enc, "=", "")
+		directURL = "https://api.onedrive.com/v1.0/shares/u!" + enc + "/root/content"
+	case driveTypeBusiness:
+		// Method: https://docs.microsoft.com/en-us/sharepoint/dev/spfx/shorter-share-link-format
+		// Example:
+		//   https://{tenant}-my.sharepoint.com/:t:/g/personal/{user_email}/{Opaque_String}
+		//   --convert to->
+		//   https://{tenant}-my.sharepoint.com/personal/{user_email}/_layouts/15/download.aspx?share={Opaque_String}
+		if len(segments) != 8 {
+			fs.Logf(f, cnvFailMsg)
+			return shareURL, nil
+		}
+		directURL = strings.Join(segments[:3], "/") + "/" + segments[5] + "/" + segments[6] +
+			"/_layouts/15/download.aspx?share=" + segments[7]
+	case driveTypeSharepoint:
+		// Method: Similar to driveTypeBusiness
+		// Example:
+		//   https://{tenant}-my.sharepoint.com/:t:/s/{site_name}/{Opaque_String}
```

Comment:
```suggestion
		//   https://{tenant}.sharepoint.com/:t:/s/{site_name}/{Opaque_String}
```

URLs for Sharepoint don't contain `-my`. (Same applies to others)